docs/source/notebooks/demo_models.py

```diff
@@ -0,0 +1,154 @@
+                              GSNodeEncoderInputLayer,
+                              RelationalGCNEncoder,
+                              EntityClassifier,
+                              ClassifyLossFunc,
+                              GSgnnLinkPredictionModel,
+                              LinkPredictDotDecoder,
+                              LinkPredictBCELossFunc)
+
+
+class RgcnNCModel(GSgnnNodeModel):
+    """ A simple RGCN model for node classification using Graphstorm APIs
+
+    This RGCN model extends GraphStorm's GSgnnNodeModel, and it has the standard GraphStorm
+    model architecture:
+    1. an input layer that converts input node features to the embeddings with hidden dimensions
+    2. a GNN encoder layer that performs the message passing work
+    3. a decoder layer, i.e., EntityClassifier, that transfors node representations into logits
+    for classification, and
+    4. a loss function that matches classification tasks.
+
+    Then the model also initialize its own optimizer object.
+
+    Arguments
+    ----------
+    g: DistGraph
+        A DGL DistGraph
+    num_hid_layers: int
+        The number of gnn layers
+    node_feat_field: dict of list of strings
+        The list features for each node type to be used in the model
+    hid_size: int
+        The dimension of hidden layers.
+    num_classes: int
+        The target number of classes for classification
+    multilabel: boolean
+        Indicator of if this is a multilabel task
+    """
+    def __init__(self,
+                 g,
+                 num_hid_layers,
+                 node_feat_field,
+                 hid_size,
+                 num_classes,
+                 multilabel=False):
+        super(RgcnNCModel, self).__init__(alpha_l2norm=0.)
+
+        # extract feature size
+        feat_size = gs.get_node_feat_size(g, node_feat_field)
+
+        # set an input layer encoder
+        encoder = GSNodeEncoderInputLayer(g=g, feat_size=feat_size, embed_size=hid_size)
+        self.set_node_input_encoder(encoder)
+
+        # set a GNN encoder
+        gnn_encoder = RelationalGCNEncoder(g=g,
+                                           h_dim=hid_size,
+                                           out_dim=hid_size,
+                                           num_hidden_layers=num_hid_layers-1)
+        self.set_gnn_encoder(gnn_encoder)
+
+        # set a decoder specific to node classification task
+        decoder = EntityClassifier(in_dim=hid_size,
+                                   num_classes=num_classes,
+                                   multilabel=multilabel)
+        self.set_decoder(decoder)
+
+        # classification loss function
+        self.set_loss_func(ClassifyLossFunc(multilabel=multilabel))
+
+        # initialize model's optimizer
+        self.init_optimizer(lr=0.001,
+                            sparse_optimizer_lr=0.01,
+                            weight_decay=0)
+
+
+class RgcnLPModel(GSgnnLinkPredictionModel):
```

Comment:
Is RgcnLPModel used in this PR? If not, let's remove it.
We can add it in the next PR.